File: ConsoleApp1/ConsoleApp1/Nguoi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
	 interface DacTrungDocGia
	{
		void GiaHanThe(Date Today);
		void NapThemTien(double Tien);
	    void DangKyTaiKhoanMoi(Date Today);
	}
	class Nguoi
    {
		protected string sHoTen;	
        protected string sDiaChi;
        protected string sSDT;
        protected Date dtNgaySinh;
		
		public string HoTen
		{
			get
			{
				return this.sHoTen;
			}
		}
		public string DiaChi
		{
			get
			{
				return this.sDiaChi;
			}
		}
		public string SDT
		{
			get
			{
				return this.sSDT;
			}
		}
		public Date NgaySinh
		{
			get
			{
				return this.dtNgaySinh;
			}
		}
		public Nguoi()
		{

		}
		public Nguoi(string HoTen, string DiaChi, string SDT, int NgaySinh, int ThangSinh,int NamSinh)
        {
            this.sHoTen = HoTen;
            this.sDiaChi = DiaChi;
            this.sSDT = SDT;
			Date Temp = new Date(NgaySinh, ThangSinh, NamSinh);
            this.dtNgaySinh = Temp;
        }
		public virtual void Nhap()
		{
			Console.Write("Ten: ");
			this.sHoTen = Console.ReadLine();
			Console.Write("Dia chi: ");
			this.sDiaChi = Console.ReadLine();
			Console.Write("SDT: ");
			this.sSDT = Console.ReadLine();
			Console.Write("Ngay sinh: ");
			this.dtNgaySinh.iNgay = Convert.ToInt32(Console.ReadLine());
			Console.Write("Thang sinh: ");
			this.dtNgaySinh.iThang = Convert.ToInt32(Console.ReadLine());
			Console.Write("Nam sinh: ");
			this.dtNgaySinh.iNam = Convert.ToInt32(Console.ReadLine());
		}
        public virtual void Xuat()
        {
            Console.WriteLine("Ho Ten: " + this.sHoTen);
            Console.WriteLine("Dia Chi: " + this.sDiaChi);
            Console.WriteLine("SDT: " + this.sSDT);
            Console.WriteLine("Ngay Sinh: " + this.dtNgaySinh.iNgay+"/"+this.dtNgaySinh.iThang+"/"+this.dtNgaySinh.iNam+" ");
        }
    }
}

File: Quan_Ly_Thu_Vien/src/Nguoi.java
import java.io.IOException;
import java.util.Scanner;
interface DacTrungDocGia
{
	 void GiaHanTheThuVien(Date A);	
	 void DangKyTaiKhoanMoi(Date Today);
}
public abstract  class Nguoi 
{

	protected String sHoTen;
	protected String sDiaChi;
	protected int iSDT;
	public String getHoTen()
	{
		return this.sHoTen;
	}
	public void setHoten(String a)
	{
		this.sHoTen = a;
	}
	public String getDiaChi()
	{
		return this.sHoTen;
	}
	public void setSDT(String a)
	{
		this.sHoTen = a;
	}
	public int getSDT()
	{
		return this.iSDT;
	}
	public void setSDT(int a)
	{
		this.iSDT = a;
	}

	public Nguoi(String HoTen,String DiaChi,int SDT) 
	{
		super();
		this.sHoTen=HoTen;
		this.sDiaChi=DiaChi;
		this.iSDT=SDT;
	}
	public Nguoi()
	{
		super();
	}
	@SuppressWarnings("resource")
	public void Nhap() throws IOException
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("Hay nhap vao ten: ");	
		this.sHoTen = sc.nextLine();
		System.out.println(" Hay nhap vao so dien thoai");
		this.iSDT=sc.nextInt();
	}
	
	public void Xuat() throws IOException
	{
		System.out.println(" Ma la: "+this.sHoTen);
		System.out.println(" SDT la: "+this.iSDT);
		System.out.println(" Dia chi la: "+this.sDiaChi);
	}
	public abstract  void Tien(double Tien);
}
